lib/filter/filter-re.c

```diff
@@ -244,13 +244,15 @@ filter_match_eval(FilterExprNode *s, LogMessage **msgs, gint num_msg)
   FilterMatch *self = (FilterMatch *) s;
 
   if (G_LIKELY(self->super.value_handle))
-    return filter_re_eval(s, msgs, num_msg);
+    self->super.super.eval = filter_re_eval;
```

Comment:
I understand we save a dispatch function during runtime, so that's a +1. Also, this makes the `G_LIKELY` unnecessary, so that's again a +1.

On the other hand, it is harder to reason about correctness. For example we need to think that the lazy update works with multi threads. That's easy to reason though. Still, I tend to think the original is easier to maintain.

Is it possible to calculate the dispatch earlier? Say in init phase? That would be the best.

Reply:
yes, it can be.

Reply:
I did not rewrite the branch, but pushed a last commit doing just that.